spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java

```diff
@@ -516,20 +526,22 @@ public void setUseRelativeRedirects(Boolean useRelativeRedirects) {
 			this.useRelativeRedirects = useRelativeRedirects;
 		}
 
+		@DeprecatedConfigurationProperty(replacement = "server.tomcat.remote-ip-valve.remote-ip-header")
 		public String getRemoteIpHeader() {
-			return this.remoteIpHeader;
+			return this.remoteIpValve.getRemoteIpHeader();
 		}
 
 		public void setRemoteIpHeader(String remoteIpHeader) {
-			this.remoteIpHeader = remoteIpHeader;
+			this.remoteIpValve.setRemoteIpHeader(remoteIpHeader);
 		}
 
+		@DeprecatedConfigurationProperty(replacement = "server.tomcat.remote-ip-valve.host-header")
 		public String getHostHeader() {
-			return this.hostHeader;
+			return this.remoteIpValve.getHostHeader() == null ? this.hostHeader : this.remoteIpValve.getHostHeader();
```

Comment:
it should be just `return this.remoteIpValve.getHostHeader();`

Reply:
Yes, fixed.